IPython/html/static/tree/js/notebooklist.js

```diff
@@ -332,104 +431,179 @@ define([
+                        that.contents.rename(path, input.val()).then(function() {
+                            that.load_list();
+                        }).catch(function(e) { 
+                            dialog.modal({
+                                title : "Error",
+                                body : $('<div/>')
+                                    .text("An error occurred while renaming \"" + path + "\" to \"" + input.val() + "\".")
+                                    .append($('<div/>').addClass('alert alert-danger').text(String(e))),
+                                buttons : {
+                                    OK : {}
+                                }
+                            });
+                        });
+                    }
+                },
+                Cancel : {}
+            },
+            open : function () {
+                // Upon ENTER, click the OK button.
+                input.keydown(function (event) {
+                    if (event.which === keyboard.keycodes.enter) {
+                        d.find('.btn-primary').first().click();
+                        return false;
                     }
                 });
-                return false;
-            });
-        item.find(".item_buttons").append(duplicate_button);
+                input.focus().select();
+            }
+        });
     };
 
-    NotebookList.prototype.add_delete_button = function (item) {
-        var notebooklist = this;
-        var delete_button = $("<button/>").text("Delete").addClass("btn btn-default btn-xs").
-            click(function (e) {
-                // $(this) is the button that was clicked.
-                var that = $(this);
-                // We use the filename from the parent list_item element's
-                // data because the outer scope's values change as we iterate through the loop.
-                var parent_item = that.parents('div.list_item');
-                var name = parent_item.data('name');
-                var path = parent_item.data('path');
-                var message = 'Are you sure you want to permanently delete the file: ' + name + '?';
-                dialog.modal({
-                    title : "Delete file",
-                    body : message,
-                    buttons : {
-                        Delete : {
-                            class: "btn-danger",
-                            click: function() {
-                                notebooklist.contents.delete(path).then(
-                                    function() {
-                                        notebooklist.notebook_deleted(path);
+    NotebookList.prototype.delete_selected = function() {
+        var message;
+        if (this.selected.length == 1) {
+            message = 'Are you sure you want to permanently delete: ' + this.selected[0].name + '?';
+        } else {
+            message = 'Are you sure you want to permanently delete the ' + this.selected.length + ' files selected?';
```

Comment:
Should this say something like '... files/folders selected?' I don't think of 'files' as including folders.